src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -1632,20 +1634,17 @@ public void add(final int index, final double mod, final String desc) {
       case STENCH_SPELL_DAMAGE:
       case SPOOKY_SPELL_DAMAGE:
       case SLEAZE_SPELL_DAMAGE:
-        String name = Modifiers.getNameFromLookup(desc);
-        if (!name.equals("Bendin' Hell") && !name.equals("Bow-Legged Swagger")) {
-          this.extras[index] += mod;
-        }
-        this.doubles[index] += mod;
-        break;
       case EXPERIENCE:
       case MUS_EXPERIENCE:
       case MYS_EXPERIENCE:
       case MOX_EXPERIENCE:
-        name = Modifiers.getNameFromLookup(desc);
-        if (!name.equals("makeshift garbage shirt")) {
-          this.extras[index] += mod;
-        }
+      case MUS_EXPERIENCE_PCT:
+      case MYS_EXPERIENCE_PCT:
+      case MOX_EXPERIENCE_PCT:
+        // Possibly doubled by makeshift garbage shirt, Bendin' Hell, Bow-Legged Swagger, or
+        // Dirty Pear.
+        // TODO: Figure out which ones aren't doubled and exclude them. BoomBox?
+        this.extras[index] += mod;
```

Comment:
sorry, I don't understand what's gone on here. I'm not familiar with this area: what are "extras"? "doubles" are modifiers which are of type `double`? Why did the previous one conditionally add to extras, but you unconditionally add?

Reply:
The previous check was never true, as it had been written incorrectly (the lookup string passed in looks like Effect[1234], not Effect[name] which is what the check was expecting). I'll add a comment to better explain how it currently works and possibly change the name.

`doubles` is (in current code) an array of all modifiers of type double - anything you would reference in ASH as a `numeric_modifier`. When two Modifiers objects are added together, typically the array on the RHS has only a few nonzero values, and the array on the LHS is used as an accumulator. `extras` is used as an accumulator for what any doubler would add, since some sources of e.g. item drop aren't subject to doubling. Later, in `recalculateAdjustments`, as you can see the code checks to see whether you have each doubler and adds the `extras` value to `doubles` if so. This handles double-doubling cleanly too, which is a nice feature of the existing code.

Reply:
Changed the name of the field.